src/plugin.validation.dns.infomaniak/InfomaniakClient.cs

```diff
@@ -0,0 +1,100 @@
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using PKISharp.WACS.Plugins.ValidationPlugins.Models;
+
+namespace PKISharp.WACS.Plugins.ValidationPlugins;
+
+internal class InfomaniakClient
+{
+    private readonly ILogService _log;
+    private readonly HttpClient _httpClient;
+    private const string Root = "https://api.infomaniak.com/";
+
+    public InfomaniakClient(string apiToken, ILogService logService, IProxyService proxyService)
+    {
+        _log = logService;
+        _httpClient = proxyService.GetHttpClient();
+        _httpClient.BaseAddress = new Uri(Root);
+        _httpClient.DefaultRequestHeaders.Add("Authorization", $"Bearer {apiToken}");
+        _httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(MediaTypeNames.Application.Json));
+    }
+    public async Task<int> GetDomainId(string domain)
+    {
+        var apiUrl = $"/1/product?service_name=domain&customer_name={domain}";
+        var response = await _httpClient.GetAsync(apiUrl);
+        var json = await response.Content.ReadAsStringAsync();
+        var domainList = JsonConvert.DeserializeObject<DomainListResponse>(json);
+
+        var domainId = domainList?.Data?.FirstOrDefault(x => x.CustomerName == domain)?.Id ?? 0;
+        if (domainId == 0)
+        {
+            _log.Error($"Infomaniak did not return record for domain: {domain}");
+        }
+
+        _log.Information($"Infomaniak found Domain ID: {domainId}");
+        return domainId;
+    }
+
+    public async Task<int> CreateRecord(int domainId, string identifier, string value, int ttl = 300)
+    {
+        if (domainId <= 0) throw new ArgumentOutOfRangeException(nameof(domainId));
+        if (ttl <= 0) throw new ArgumentOutOfRangeException(nameof(ttl));
+
+        var apiUrl = $"/1/domain/{domainId}/dns/record";
+        var postData = new DomainRecordCreateRequest
+        {
+            Type = "TXT",
+            Source = identifier,
+            Target = value,
+            TimeToLive = ttl
+        };
+        var jsonContent = JsonConvert.SerializeObject(postData);
+        var response = await _httpClient.PostAsync(apiUrl, new StringContent(jsonContent, Encoding.UTF8, MediaTypeNames.Application.Json));
+        if (!response.IsSuccessStatusCode)
+        {
+            _log.Error("Infomaniak did not create record. Request Failure.");
+            return 0;
+        }
+
+        //Get new Record ID - used for deleting
+        var jsonResponse = await response.Content.ReadAsStringAsync();
+        var createResponse = JsonConvert.DeserializeObject<DomainRecordCreateResponse>(jsonResponse);
+        var id = (createResponse?.Data ?? 0);
+        if (id == 0)
+        {
+            _log.Error($"Infomaniak did not create record: {jsonResponse}");
+            return 0;
+        }
+        _log.Information($"Infomaniak created TXT Record with ID: {id}");
+
+        // Record take 2-3 seconds to be accessible over internet
+        await Task.Delay(5_000);
```

Comment:
Inserting a magic delay is not neccesary, the main program will wait for the records to become visible from the internet.